functions/visualizations.py

```diff
@@ -35,15 +36,42 @@ def vis_preprocessing(event: dict, context: BackgroundContext):
         session.commit()
 
 
-def _get_file_from_gcs(bucket_name: str, object_name: str) -> BytesIO:
+def _get_data_file(
+    object_name: str, as_string: bool = False
+) -> Union[BytesIO, StringIO]:
     """Download data from GCS to a byte stream and return it."""
     storage_client = storage.Client()
-    bucket = storage_client.get_bucket(bucket_name)
+    bucket = storage_client.get_bucket(GOOGLE_DATA_BUCKET)
     blob = bucket.get_blob(object_name)
     file_str = blob.download_as_string()
+    if as_string:
+        return StringIO(file_str)
     return BytesIO(bytes(file_str))
 
 
+def _get_metadata_df(trial_id: str) -> pd.DataFrame:
+    """
+    Build a dataframe containing the participant/sample metadata for this trial,
+    joined on CIMAC ID and indexed on CIMAC ID.
+    """
+    participants_blob = _get_data_file(f"{trial_id}/participants.csv", as_string=True)
+    samples_blob = _get_data_file(f"{trial_id}/samples.csv", as_string=True)
+
+    participants_df = pd.read_csv(participants_blob)
+    samples_df = pd.read_csv(samples_blob)
```

Comment:
just a thought - we also have this data in the json format in the DB, so maybe it's faster to get it from there than from GCS?

Reply:
Reviewing the `patient_sample_update.generate_csvs` cloud function where these files are created, we don't appear to be storing the contents of these files as JSON in the database - only metadata about them: https://github.com/CIMAC-CIDC/cidc-cloud-functions/blob/d72a54989fe5939d191b8f2da69441fce13b90f6/functions/patient_sample_update.py#L31

Reply:
I meant that we have the JSON that we generated this csv from, so wouldn't it be faster to call `generate` on the fly rather then reading csv from gcs

Reply:
Ah, I see - almost definitely. We can put a ticket in the backlog about this.